app/src/main/java/com/jerboa/model/SiteViewModel.kt

```diff
@@ -73,7 +74,7 @@ class SiteViewModel : ViewModel() {
         }
     }
 
-    fun getUnreadCountTotal(): Int {
+    fun getUnreadCountTotal(unreadCountRes: ApiState<GetUnreadCountResponse>): Int {
```

Comment:
How about instead of using that `derivedState` remember in each of the activities, you just turn this one from a function, into another var alongside `unreadCountRes`, and update that var in the viewmodel fetches.

Reply:
I did something better imo, i just moved the derived into the viewmodel. I tested it and it still works